compaction/compaction_manager.cc

```diff
@@ -1127,12 +1127,23 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
-    do_stop();
-    if (auto cm = std::exchange(_task_manager_module, nullptr)) {
-        co_await cm->stop();
-    }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    try {
+        do_stop();
+        if (auto cm = std::exchange(_task_manager_module, nullptr)) {
+            auto f = cm->stop();
+            if (!_stop_future) {
+                _stop_future = std::move(f);
+            } else {
+                _stop_future = f.finally([sf = std::move(*_stop_future)] () mutable {
+                    return std::move(sf);
+                });
+            }
+        }
+        if (_stop_future) {
+            co_await std::move(*_stop_future);
+        }
+    } catch (...) {
+        cmlog.warn("Ignored error while stopping the manager: {}", std::current_exception());
     }
 }
```

Comment:
If all we're waiting for here is task_manager and compaction_manager, then the error handling should be moved there.